presto-main/src/main/java/com/facebook/presto/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -73,15 +77,27 @@ public DistributedExecutionPlanner(SplitManager splitManager)
 
     public StageExecutionPlan plan(SubPlan root, Session session)
     {
+        Visitor visitor = new Visitor(session);
+        try {
+            return plan(root, visitor);
+        }
+        catch (Throwable t) {
+            visitor.cleanup();
```

Comment:
I'd call this something other than `cleanup` because that sounds like it should always be called.